File: Python/lab18/ControllSystem.py
from Order import Order
from Version import Version


class ControllSystem:
    def __init__(self, order):
        self.order = order
        self.version = Version()

    def get_order(self):
        return self.order

    def set_order(self, name, type):
        self.order.set_name(name)
        self.order.set_type(type)

    def save_order_to_memento(self):
        self.version.add_version(self.order.copy())

    def restore_version_from_memento(self, version):
        self.order = self.version.get_version(version)

File: Java/lab18/src/main/java/org/example/ControllSystem.java
package org.example;

public class ControllSystem {
    private Order order;

    public ControllSystem(Order order) {
        this.order = order;
    }

    private final Version version=new Version();

    public Order getOrder() {
        return order;
    }

    public void setOrder(String name,String type) {
        this.order.setName(name);
        this.order.setType(type);
    }

    public void saveOrderToMemento(){
        version.addVersion(order.copy());
    }
    public void restoreVersionFromMemento(int version){
        order= this.version.getVersion(version);
    }
}
